client/src/js/todos/dragAndDrop.js

```diff
@@ -0,0 +1,154 @@
+import { $, $$, closest } from '../util.js';
+
+export class DragAndDrop {
+  constructor() {
+    this.target = document;
+
+    this.isDragging = null;
+    this.isClone = false;
+
+    this.dragCard = null;
+    this.clonedDragCard = null;
+    this.cardPositionInfo = [];
+    this.moveCardState = false;
+  }
+
+  init() {
+    this.target.addEventListener('mousedown', this.mousedownEventHandler);
+    this.target.addEventListener('mouseup', this.mouseupEventHandler);
+    this.target.addEventListener('mousemove', this.mousemoveEventHandler);
+  }
+
+  mousedownEventHandler = (e) => {
+    if (closest('.default', e.target)) {
+      this.isDragging = true;
+      this.dragCard = closest('.default', e.target);
+      this.column = closest('.column-list', e.target);
+    }
+  };
+
+  mouseupEventHandler = (e) => {
+    this.isDragging = false;
+    this.moveCardState = false;
+    if (this.isClone) {
+      this.clonedDragCard.remove();
+      this.dragCard.classList.replace('place', 'default');
+      this.isClone = null;
+      this.cardPositionInfo = [];
+    }
+  };
+
+  mousemoveEventHandler = (e) => {
+    if (!this.isDragging) return;
+
+    if (!this.isClone) {
+      this.setCardsPosition();
+      this.cloneCard(e);
+    }
+
+    const mouseX = e.clientX;
+    const mouseY = e.clientY;
+    this.clonedDragCard.style.left = mouseX + 'px';
+    this.clonedDragCard.style.top = mouseY + 'px';
+
+    const cloneCardPosData = this.clonedDragCard.getBoundingClientRect();
+    const columnPosData = this.column.getBoundingClientRect();
+
+    const draggedX = cloneCardPosData.x;
+    const draggedY = cloneCardPosData.y;
+
+    if (columnPosData.x - cloneCardPosData.x > cloneCardPosData.width / 2) {
```

Comment:
if문의 경우 조건문만 따로 변수로 빼주어도 가독성이 많이 올라가는 효과를 볼 수 있습니다!  
```suggestion
    // ex
    const topPosition = columnPosData.x - cloneCardPosData.x > cloneCardPosData.width / 2;
    if (topPosition) {
```